proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,176 @@
+      
+  2. Parameter is an array of enum types (see example 2)
+      SDL core has to remove unknown value from the array.
+      A case is possible when all values are unknown and thus get removed.
+      In this case, SDL core has to proceed the same way as in case 1.
+      
+  3. Parameter is part of the structure (see example 3)
+      SDL has to proceed the same way as in case #1 or #2 and remove this parameter from the structure.
+      However, if the parameter is mandatory the structure becomes invalid.
+      In this case, SDL core has to remove the whole structure from the request.
+      During this process, SDL has to proceed recursively from the very bottom level up to the top.
+
+  4. Parameter is a part of the structure which is a part of an array (see example 4)
+      SDL core has to process it the same way as in case #3 and remove the structure as an item of the array.
+  Once all the parameters are processed SDL core has to proceed with the request as usual.
+  If at least one value of at least one parameter was cut off, SDL core has to update the response to Mobile Application as follows:
+
+  1. If the response was processed successfully, SDL core has to provide the `WARNINGS` result code instead of `SUCCESS`.
+  2. SDL core has to provide removed enum items in `info` string of the response.
+      Since there could be more than one parameter with cut-off value `info` message can be constructed the following way:
+      `Invalid enums were removed: <param_1>:<enum_value_1>,<enum_value_2>;<param_2>:<enum_value_3> ...`
+      If `info` parameter contains other value belonging to the original processing result SDL core has to append information about cut-off enum value(s) to the existing value.
+      
+  **Examples:**
+  In each example below the Mobile Application sends a request to SDL core.
+  1. Request contains a parameter which value is an enum item
+      Request: `SetMediaClockTimer`
+      
+      1.1 Parameter is mandatory: `updateMode="UNKNOWN"`
+      SDL core removes the item `UNKNOWN` from the request.
+      The value of `updateMode` parameter becomes empty.
+      SDL omits empty parameter from the request.
+      Since parameter is mandatory (according to Mobile API), the request fails with the following response to Mobile Application:
+      `success=false, resultCode = "INVALID_DATA", info="Invalid enums were removed: updateMode:UNKNOWN"`
+      
+      1.2 Parameter is optional: `audioStreamingIndicator="UNKNOWN"`
+      SDL core removes the `UNKNOWN` item from the request.
+      The value of `audioStreamingIndicator` parameter becomes empty.
+      SDL core omits the empty parameter in the request.
+      Since the parameter is optional (according to Mobile API), the request is successfully processed with the following response to Mobile Application:
+      `success=true, resultCode = "WARNINGS", info="Invalid enums were removed: audioStreamingIndicator:UNKNOWN"`
```

Comment:
```suggestion
      1.2 Parameter is optional: `audioStreamingIndicator="UNKNOWN"`
    
      SDL Core removes the `UNKNOWN` item from the request.
      The value of `audioStreamingIndicator` parameter becomes empty.
      SDL Core omits the empty parameter in the request.
      Since the parameter is optional (according to Mobile API), the request is successfully processed with the following response to mobile application:
      `success=true, resultCode = "WARNINGS", info="Invalid enums were removed: audioStreamingIndicator:UNKNOWN"`
```
Please also see comment above regarding line breaks.